tests/CSharpTemp/CSharpTemp.Tests.cs

```diff
@@ -344,4 +345,34 @@ public void TestNullAttributedFunctionPtr()
         }
     }
 
+    [Test]
+    public unsafe void TestMarshallingOfCharPtr()
+    {
+        var obj = new CheckMarshllingOfCharPtr();
+        var wstr = obj.FuncRetWideCharPtr;
+        Assert.AreEqual('S', wstr[0]);
+        var wstrBuilt = new StringBuilder(new string(wstr));
+        wstrBuilt[0] = 'j';
+        obj.FuncWithWideCharPtr(wstrBuilt);
+        wstr = obj.FuncRetWideCharPtr;
+        Assert.AreEqual('j', wstr[0]);
+
+        var obj2 = new CheckMarshllingOfCharPtr();
+        var str = obj2.FuncRetCharPtr;
+        byte[] bytes = Encoding.UTF32.GetBytes(new string(str));
+        var myString = Encoding.ASCII.GetString(bytes);
```

Comment:
The correct way to do it is with Marshal.PtrToStringUni:

    var myString = Marshal.PtrToStringUni(obj2.FuncRetCharPtr);

Let me know if you have problems with it.